includes/AMP/Sanitization.php

```diff
@@ -0,0 +1,459 @@
+		 *
+		 * {@link https://amp.dev/documentation/guides-and-tutorials/optimize-and-measure/optimize_amp/ AMP Hosting Guide}
+		 */
+		foreach ( Amp::RENDER_DELAYING_EXTENSIONS as $extension ) {
+			if ( isset( $amp_scripts[ $extension ] ) ) {
+				$ordered_scripts[ $extension ] = $amp_scripts[ $extension ];
+				unset( $amp_scripts[ $extension ] );
+			}
+		}
+
+		/*
+		 * "7. Specify the <script> tags for remaining extensions (e.g., amp-bind ...). These extensions are not render-delaying
+		 * and therefore should not be preloaded as they might take away important bandwidth for the initial render."
+		 */
+		ksort( $amp_scripts );
+		$ordered_scripts = array_merge( $ordered_scripts, $amp_scripts );
+		/**
+		 * Script element.
+		 *
+		 * @var DOMElement $ordered_script
+		 */
+		foreach ( $ordered_scripts as $ordered_script ) {
+			if ( $previous_node && $previous_node->nextSibling ) {
+				$document->head->insertBefore( $ordered_script, $previous_node->nextSibling );
+				$previous_node = $ordered_script;
+			}
+		}
+	}
+
+	/**
+	 * Returns AMP extension URLs, keyed by extension name.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @return array List of extensions and their URLs.
+	 */
+	protected function get_extension_sources() {
+		$specs = [];
+		// Register all AMP components as defined in the spec.
+		foreach ( AMP_Allowed_Tags_Generated::get_extension_specs() as $extension_name => $extension_spec ) {
+			$src = sprintf(
+				'https://cdn.ampproject.org/v0/%s-%s.js',
+				$extension_name,
+				end( $extension_spec['version'] )
+			);
+
+			$specs[ $extension_name ] = $src;
+		}
+
+		return $specs;
+	}
+
+	/**
+	 * Determine whether AMP dev mode is enabled.
+	 *
+	 * When enabled, the <html> element will get the data-ampdevmode attribute and the plugin will add the same attribute
+	 * to elements associated with the admin bar and other elements that are provided by the `amp_dev_mode_element_xpaths`
+	 * filter.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @return bool Whether AMP dev mode is enabled.
+	 */
+	protected function is_amp_dev_mode() {
+		/**
+		 * Filters whether AMP dev mode is enabled.
+		 *
+		 * When enabled, the data-ampdevmode attribute will be added to the document element and it will allow the
+		 * attributes to be added to the admin bar. It will also add the attribute to all elements which match the
+		 * queries for the expressions returned by the 'web_stories_amp_dev_mode_element_xpaths' filter.
+		 *
+		 * @since 1.1.0
+		 *
+		 * @param bool Whether AMP dev mode is enabled.
+		 */
+		return apply_filters(
+			'web_stories_amp_dev_mode_enabled',
+			(
+				// For the few sites that forcibly show the admin bar even when the user is logged out, only enable dev
+				// mode if the user is actually logged in. This prevents the dev mode from being served to crawlers
+				// when they index the AMP version.
+				( is_admin_bar_showing() && is_user_logged_in() )
+			)
+		);
```

Comment:
```suggestion
		// For the few sites that forcibly show the admin bar even when the user is logged out, only enable dev
		// mode if the user is actually logged in. This prevents the dev mode from being served to crawlers
		// when they index the AMP version.
				
		$debug = ( is_admin_bar_showing() && is_user_logged_in() );
		/**
		 * Filters whether AMP dev mode is enabled.
		 *
		 * When enabled, the data-ampdevmode attribute will be added to the document element and it will allow the
		 * attributes to be added to the admin bar. It will also add the attribute to all elements which match the
		 * queries for the expressions returned by the 'web_stories_amp_dev_mode_element_xpaths' filter.
		 *
		 * @since 1.1.0
		 *
		 * @param bool Whether AMP dev mode is enabled.
		 */
		return apply_filters( 'web_stories_amp_dev_mode_enabled', $debug );
```